### PR TITLE
Added tests for the views

### PR DESCRIPTION
**Pull request by Dagic-zewdu**
- Tests for Login page
- Tests for User index page
- Tests for user show page
- Tests for User post index page
- Tests for the Post-show page

### CODE REVIEW

**Review by Deepakdanger — CHANGES_REQUESTED** [review]
Hi @Dagic-zewdu & @muneebulrehman, 

Good job so far! 👏
There are some issues that you still need to work on to go to the next project but you are almost there!

## To Highlight :100: 
- No linter error :heavy_check_mark: 

## Required Changes ♻️

- [ ] All your tests are failing. 

Screenshot from 2022-06-16 21-49-49

Screenshot from 2022-06-16 21-50-01

I think it's happening because of this error(attached below). Whatever we are giving data to **photo** column, they are trying to fetch in the asset pipeline. Because of this, we are unable to go to any of the pages. Please fix it.

Screenshot from 2022-06-16 21-57-09

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by julie-ify — COMMENTED** [review]
Hi <STUDENT_MENTION>, 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

### You have submitted a Pull Request with failing linters

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

File: spec/features/post_show_spec.rb .rb [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+  end
+
+  it 'shows how many comments the post has' do
+    expect(page).to have_content '2 Comments'
+  end
+
+  it 'shows how many likes the post has' do
+    expect(page).to have_button '0 Likes'
+  end
+end
```
Comment (ahmetbozaci) [inline spec/features/post_show_spec.rb .rb]: - [ ] Kindly fix the error.

image

File: spec/features/user_index_spec.rb [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+    expect(page).to have_content 'Number of posts: 1'
+    expect(page).to have_content 'Number of posts: 5'
+  end
+
+  scenario 'gets redirected to user page after click on it' do
+    click_link @name1
+
+    expect(page).to have_current_path(user_path(@user1))
+  end
+end
```
Comment (ahmetbozaci) [inline spec/features/user_index_spec.rb]: - [ ] Kindly fix the error.
image

File: spec/features/user_show_spec.rb [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+
+    expect(page).to have_current_path(user_post_path(@user, @user.posts.second))
+  end
+
+  it 'gets redirected to all posts page when user clicks on see all posts button' do
+    click_link 'See all posts'
+
+    expect(page).to have_current_path(user_posts_path(@user))
+  end
+end
```
Comment (ahmetbozaci) [inline spec/features/user_show_spec.rb]: - [ ] Kindly fix the error.

image

**Review by ahmetbozaci — CHANGES_REQUESTED** [review]
# STATUS: Changes Required  ♻️


@Dagic-zewdu  ✋🏼 
#### Great work on making the changes requested by a previous reviewer 👏 
#### You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.
___
### ♻️ Request changes ♻️

- [ ] _Check the comments under the review._
- [ ] Your branches' names are not descriptive. This is a [general](https://github.com/microverseinc/curriculum-transversal-skills/blob/main/documentation/articles/professional_repo_rules.md#:~:text=Respect%20the%20rules%20of%20the%20code%20review%20flow%20that%20your%20team%20follows.%20Use%20descriptive%20names%20for%20your%20branches%20(e.g.%2C%20refactor%2Dauthentication%2C%20user%2Dcontent%2Dcache%2Dkey%2C%20make%2Dretina%2Davatars)%20and%20make%20sure%20that%20your%20%27feature%20branch%27%20is%20up%2Dto%2Ddate%20with%20the%20%27main%20branch%27%20in%20your%20repository.) rule for every project. Please pay attention to this in your next projects.

___

### Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._
___

### Cheers and Happy coding! 💻 👏
almost


___

_Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification._

_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._
___

_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form.](https://airtable.com/shrQAqnBwek5a0O0s)_

[GitHub](https://github.com/ahmetbozaci) | [LinkedIn](https://linkedin.com/in/ahmetbozaci) | [Twitter](https://twitter.com/ahmtbozaci)

_Reviewed by [Ahmet Bozacı ](https://github.com/ahmetbozaci)_

**Review by Mre55 — APPROVED** [review]
**Hi @Dagic-zewdu,**

# Status: APPROVED 🏆 🎉 🎊 

Your project is complete!  There is nothing else to say other than... it's time to merge it  🚀: 
Congratulations! 🎉 

## To highlight 🎯

- The PR title and description are professional and descriptive ✅ 
- All tests are passing ✅
- The commit messages are meaningful ✅ 
- Linters are working properly ✅ 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**